app/controllers/newflow/login_signup_controller.rb

```diff
@@ -176,7 +176,9 @@ def oauth_callback
 
             if !user.activated?
               # not activated means signup
-              save_unverified_user(user)
+              unverified_user = EnsureUnverifiedUser.call(user).outputs.user
```

Comment:
law of demeter violation.  spanning multiple types with periods. https://medium.com/@gioch/design-patterns-law-of-demeter-with-rails-49a44a9689fe

Violations like this dont need to be "the law".  just common sense.  when one call spans multiple types, it's creating a coupling that is brittle.  it makes changing other code down the road way more difficult.

Reply:
Yes, I'm aware of the law of Demeter and that I'm breaking that rule there when I do that. The caveat is that you should know when to break the rule(s). In this case, and with all usage of the Lev gem in Accounts, we trust that Lev is pretty stable, relatively stable and so its interface will not change.

I did go ahead and encapsulated that piece of code in a function so that the controller action is more easily testable though. 👍 . Still not perfect but still an improvement, I believe.